pontoon/static/js/widgets/errors.js

```diff
@@ -0,0 +1,23 @@
+
+import React from 'react';
+
+
+export default class ErrorList extends React.Component {
```

Comment:
I've seen several code examples where people use functions for components that don't need lifecycle stuff. This would be a good fit. What do you think of that?

Reply:
sounds like a good idea, but i would need to investigate or be shown an example to be sure i follow

Reply:
That'd look like 
```javascript
const ErrorList = ({ errors }) => {
    // This is basically the render() function.
    return <ul></ul>
};
export default ErrorList;
```

It's being used in the React docs, like here: https://reactjs.org/docs/components-and-props.html

Reply:
im back to the why use a function question. Functional programming feels to me like a javascript throwback to a time when it didnt have the basics of object-oriented programming. It  also makes it difficult to extend the errorlist in any way - something i can certainly imagine doing

Reply:
Ive switched to using a `PureComponent` which i believe provides the same benefits as a stateless function, but with a class structure